source/plugin/economy/practices.rst

```diff
@@ -0,0 +1,56 @@
+==========================
+Economy API Best Practices
+==========================
+
+The Economy API tries to be abstract enough to give economy plugins flexibility in how they operate.
+In order to give economy plugins as much control as possible, plugins consuming the Economy API should
+follow some guidelines when working with it:
+
+Withdrawing money
+=================
+
+Plugins should *not* check if an account has enough money before attempting to withdraw it. While this may
+sound counter-intuitive, it allows economy plugins to fully control how they handle negative balances.
+
+By checking yourself if the account has enough money, you prevent the economy plugin from (potentially)
+allowing a negative balance. For example, one economy plugin might want to allow negative balances to admins,
+or players with a certain permission. By performign the check yourself, you take this power away from the economy plugin.
+
+This code illustrates what **not** to do:
+
+.. code-block:: java
+
+        EconomyService service = ...
+        Account account = ...
+        BigDecimal requiredAmount = BigDecimal.valueOf(20);
+
+        // BAD: Don't perform this check
+        if (account.getBalance(service.getDefaultCurrency()).compareTo(requiredAmount) == -1) {
```

Comment:
better use < 0 here. While the contract for `BigDecimal#compareTo(BigDecimal)` guarantees that only -1, 0 or 1 are returned, that is not necessarily true for any `Comparable`